artio-codecs/src/main/java/uk/co/real_logic/artio/dictionary/generation/AcceptorGenerator.java

```diff
@@ -152,9 +152,10 @@ private void generateDecoder()
 
             generateDecoderOnMessage(decoderOutput);
 
-            for (final Message message : dictionary.messages())
```

Comment:
Hi Richard - Thanks for the feedback. I've implemented the changes you requested. As for replacing the foreach loop with a counting loop - the index is passed to ``generateDecoderCase`` so that when index != 0 we can generate an else to go with the if (given that we are no longer switching)

```
    private void generateDecoderCase(
        final Writer decoderOutput,
        final Message message,
        final int index) throws IOException
    {
        decoderOutput.append(String.format(
            "        %1$sif (messageType == %2$s.MESSAGE_TYPE)\n" +
            "        {" +
            "            %3$s.decode(buffer, offset, length);\n" +
            "            acceptor.on%4$s(%3$s);\n" +
            "            %3$s.reset();\n" +
            "        }\n\n",
            index == 0 ? "" : "else ",
            decoderClassName(message),
            formatPropertyName(message.name()),
            message.name()
        ));
    }```